Source/ChocolateyGui.Common/Services/ConfigService.cs

```diff
@@ -147,13 +222,14 @@ public void EnableFeature(ChocolateyGuiConfiguration configuration)
 
         public void DisableFeature(ChocolateyGuiConfiguration configuration)
         {
+            var chosenAppConfiguration = configuration.FeatureCommand.Global ? _globalAppConfiguration : _userAppConfiguration;
             var featureProperty = GetProperty(configuration.FeatureCommand.Name, true);
-            var featureValue = GetPropertyValue<bool>(_appConfiguration, featureProperty);
+            var featureValue = GetPropertyValue<bool?>(chosenAppConfiguration, featureProperty);

```

Comment:
I have been wondering, maybe an alternative aproach here could be taken instead of using a nullable bool?

Like for instance passing having an overload that allows a default value to return if the actual value is null.

Reply:
I'm not a C# person, but from a PL perspective I'd rather not "re-introduce" nullable types whenever possible.

In general, I'd prefer a more simplistic approach such as only storing default/global values in the global db and only have user-specific values in the user db.
* if you use the `--global` switch, set the value in global db and remove the key/value pair from the users db ..
* if the user changes a setting, only set it in the user db.
whenever accessing a config value, use some proxy method that first checks if the key exists in the user database, and return either that or the global attribute (if no user-specific setting is present)

That said - again, I'm really not a C# person - just don't like nullable types :-D

Reply:
@AdmiringWorm @mwallner sorry for not replying to either of you sooner about this.  I have just come back around to look at this again, and I am going to proceed with the current implementation.  I believe that this provides the functionality that is required, and covers all the edge cases regarding whether a global default value has been set or not, without over complicating the matter.

I have just taken this for a spin again this morning, alongside the Chocolatey GUI Licensed Extension, and I am happy to say that it is working as expected, so for now, I am going to get this merged.